build.xml

```diff
@@ -142,7 +148,29 @@
 
   <!-- PHP CodeSniffer -->
   <target name="phpcbf">
-    <exec command="${srcdir}/vendor/bin/phpcbf --standard=${srcdir}/tests/phpcs.xml" escape="false" passthru="true" checkreturn="true" />
+    <!-- Run phpcs first to find any problems (faster than phpcbf): -->
+    <exec command="${srcdir}/vendor/bin/phpcs --cache=${srcdir}/tests/phpcs.cache.json --standard=${srcdir}/tests/phpcs.xml" escape="false" returnProperty="phpcs_retval" outputProperty="phpcs_output" />
+    <if>
+      <not><equals arg1="${phpcs_retval}" arg2="0" /></not>
+      <then>
+        <echo msg="Trying to fix the following issues: " />
+        <echo msg="${phpcs_output}" />
+        <!-- Extract file names from phpcs output: -->
+        <property name="phpcbf_filelist" value="${phpcs_output}">
+          <filterchain>
+            <linecontainsregexp>
+              <regexp pattern="^FILE: (.*)" />
+            </linecontainsregexp>
+            <striplinebreaks />
+            <replaceregexp>
+              <regexp pattern="FILE: " replace=" " />
+            </replaceregexp>
+          </filterchain>
+        </property>
+        <!-- Finally, run phpcbf: -->
+        <exec command="${srcdir}/vendor/bin/phpcbf --standard=${srcdir}/tests/phpcs.xml ${phpcbf_filelist}" escape="false" passthru="true" checkreturn="true" />
```

Comment:
I think we might want to change checkreturn to "false" here -- otherwise, when phpcbf fixes something, the process behaves as though something has gone wrong due to the return status. Any objection?

Reply:
No objection. It just was that way before and I didn't touch it or give it a thought at all. Changed in the last commit.